platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/internal/AbstractTaskSchedulerBuilder.java

```diff
@@ -130,12 +130,31 @@ public AbstractTaskSchedulerBuilder<OUT> configure(@NonNull final TaskSchedulerC
         return this;
     }
 
+    /**
+     * Ensures that direct schedulers do not have an unhandled task capacity set.
+     *
+     * <p>If the scheduler type is {@link TaskSchedulerType#DIRECT} or {@link TaskSchedulerType#DIRECT_THREADSAFE}
+     * and the unhandled task capacity is not 1, an {@link IllegalArgumentException} is thrown.
+     *
+     * @param type the type of the task scheduler
+     * @param unhandledTaskCapacity the unhandled task capacity
+     * @throws IllegalArgumentException if the type is direct or direct threadsafe and the unhandled task capacity is not 1
+     */
+    public static void validateDirectScheduler(
+            @NonNull final TaskSchedulerType type, final long unhandledTaskCapacity) {
+        if ((type == TaskSchedulerType.DIRECT || type == TaskSchedulerType.DIRECT_THREADSAFE)
+                && unhandledTaskCapacity != 1) {
+            throw new IllegalArgumentException("Direct schedulers cannot have an unhandled task capacity.");
+        }
+    }
```

Comment:
It seems a bit odd to make this method public. Outside potential unit tests, I don't think there will ever be use cases for somebody to call into this directly.

One way to do this differently might be to make this non-static and private. This would allow you to get rid of the arguments. If you do this, I'd rename it to something like `validateConfiguration()`, since it would be a decent place to do other checks if they ever get added.

Reply:
The method **validateDirectScheduler()** is called here: 
![image](https://github.com/user-attachments/assets/4b666d74-b3a9-4b31-97d0-2271dab66bdc)
and here: 
![image](https://github.com/user-attachments/assets/5c37bb77-43d5-4bb9-a78a-0981040db7ab)
Making it private would render it inaccessible in this build logic. Should this logic be inlined within the build() method for the classes mentioned above? Would this be a correct implementation?

Reply:
> It seems a bit odd to make this method public. Outside potential unit tests, I don't think there will ever be use cases for somebody to call into this directly.
> 
> One way to do this differently might be to make this non-static and private. This would allow you to get rid of the arguments. If you do this, I'd rename it to something like `validateConfiguration()`, since it would be a decent place to do other checks if they ever get added.

The method **validateDirectScheduler()** is called here: 
![image](https://github.com/user-attachments/assets/4b666d74-b3a9-4b31-97d0-2271dab66bdc)
and here: 
![image](https://github.com/user-attachments/assets/5c37bb77-43d5-4bb9-a78a-0981040db7ab)
Making it private would render it inaccessible in this build logic. Should this logic be inlined within the build() method for the classes mentioned above? Would this be a correct implementation?

Reply:
Perhaps this could be a private static method that lives inside `StandardTaskSchedulerBuilder.java`.

Reply:
> Perhaps this could be a private static method that lives inside `StandardTaskSchedulerBuilder.java`.

Making the method static would require the variables **type** and **unhandledTaskCapacity** in **_AbstractTaskSchedulerBuilder_** to be made static. Is this feasible?